charms/worker/k8s/src/charm.py

```diff
@@ -577,51 +595,76 @@ def _reconcile(self, event: ops.EventBase):
             self._copy_internal_kubeconfig()
             self._expose_ports()
 
-    @on_error(
-        ops.WaitingStatus("Cluster not yet ready"),
-        AssertionError,
-        subprocess.CalledProcessError,
-        InvalidResponseError,
-        K8sdConnectionError,
-    )
     def _update_status(self):
         """Check k8s snap status."""
-        if self.is_dying:
-            status.add(ops.WaitingStatus("Preparing to leave cluster"))
-            return
```

Comment:
the `death_handler` sets to `WaitingStatus("Awaiting cluster removal")`.  This wasn't necessary anymore